caddyhttp/httpserver/replacer.go

```diff
@@ -375,6 +376,26 @@ func (r *replacer) getSubstitution(key string) string {
 		}
 		elapsedDuration := time.Since(r.responseRecorder.start)
 		return strconv.FormatInt(convertToMilliseconds(elapsedDuration), 10)
+	case "{ssl_protocol}":
+		if r.request.TLS != nil {
+			for k, v := range caddytls.SupportedProtocols {
+				if v == r.request.TLS.Version {
+					return k
+				}
+			}
+			return "yes" // this should never happen, but guard in case
+		}
+		return "none" // because not using a secure channel
+	case "{ssl_cipher}":
```

Comment:
Change to `tls_cipher` or perhaps `{tls_ciphersuite}` since technically it is not a cipher, it is a cipher suite. Maybe it doesn't matter though.

Reply:
I don’t think it really matters (and it was chosen to maintain some sense of consistency with mod_ssl’s use of ```SSL_CIPHER```).

BTW, if you look at the documentation for the ```tls``` directive and its  ```ciphers``` keyword, that refers to ciphers and not cipher suites... but I’m guessing that anybody mucking around at that level knows the technicalities.

Reply:
Fair enough, this is fine. :) But rename to `tls_`.